okhttp/src/test/kotlin/com/connectrpc/okhttp/MockWebServerTests.kt

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022-2023 The Connect Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.connectrpc.okhttp
+
+import com.connectrpc.ProtocolClientConfig
+import com.connectrpc.RequestCompression
+import com.connectrpc.compression.GzipCompressionPool
+import com.connectrpc.conformance.v1.TestServiceClient
+import com.connectrpc.conformance.v1.simpleRequest
+import com.connectrpc.extensions.GoogleJavaProtobufStrategy
+import com.connectrpc.impl.ProtocolClient
+import com.connectrpc.protocols.NetworkProtocol
+import kotlinx.coroutines.test.runTest
+import okhttp3.OkHttpClient
+import okhttp3.Protocol
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import okio.Buffer
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+
+class MockWebServerTests {
+
+    @Test
+    fun `compressed empty failure response is parsed correctly`() = runTest {
+        val mockWebServer = MockWebServer()
+        mockWebServer.start()
+
+        mockWebServer.enqueue(
+            MockResponse().apply {
+                addHeader("content-encoding", "gzip")
+                addHeader("content-type", "application/json")
+                setBody(Buffer())
+                setResponseCode(401)
```

Comment:
This isn't quite representative of the original scenario we were seeing: the server was returning a non-empty response, but by the time it got [here](https://github.com/connectrpc/connect-kotlin/blob/7b7c746352c83bed44cb4659f6ba5566750c3e5b/library/src/main/kotlin/com/connectrpc/protocols/ConnectInterceptor.kt#L104), the response message buffer was already empty. 

Our hypothesis after stepping through the code a few times, was that `parseConnectUnaryException()` was consuming the message buffer before the decompress call had a change to consume it: the message was always non-empty when `parseConnectUnaryException` was called. We thought the root cause might be to decompress the response _first_, then attempt to parse it into an error message. I haven't been able to reproduce that exact scenario in this test, though; the _actual_ root cause may still be TBD.